### DISCUSSION TITLE
报错了  怎么解决Prop `className` did not match. Server: "sc-iBPTVF lhPANS" Client: "sc-hBEZvw bTffbp"     at div

### DISCUSSION THREAD

**Discussion by 2010363329** [vercel/next.js#58919] [Help] [answered]
### Summary

我想知道为什么会出现这个错误 以及怎么解决

**Comment by 2010363329** [comment]
这个错误什么玩意 怎么一会儿报错 一会儿不报错呢 为什么我感觉next.js乱七八糟的小错误一堆呢 感觉有点XX的感觉 拍错也不好排 好难用啊

**Comment by icyJoseph** [comment, marked as answer]
If you are using styled components, you need to tell the compiler to create stable classNames, so that they don't change between client and server:

- https://nextjs.org/docs/architecture/nextjs-compiler#styled-components

```js
next.config.js
module.exports = {
  compiler: {
    styledComponents:  {
      ssr: true,
    },
  },
}
```

Does that help out? You might also have other issues that are producing different class names in the browser and server, and this is a problem with the styling solution you are using, not Next.js or React.

**Comment by 2010363329** [comment]
我爱你 好人一生平安
